docs/release.md

```diff
@@ -0,0 +1,78 @@
+# Release
+
+> [!NOTE]
+> This document is based on the initial learnings from the release process and is therefore subject to change.
+
+## Preparing the release
+
+For each release, a milestone needs to be in place. All PRs that are part of the release need to be assigned to 
+the milestone.
+
+This will also make it easier to filter for the correct commits, that should be cherry-picked later.
```

Comment:
Regarding filtering, I guess it's better to have a Label assigned to The PRs, 
So the user can check the history and can apply those PRs to The release branch. 
What do you think?

Reply:
milestones are good filters, aren't they? I would prefer to not always maintain the labels. Milestones can just be closed